taskRequests/details/script.js

```diff
@@ -0,0 +1,236 @@
+const API_BASE_URL = window.API_BASE_URL;
+
+const taskRequestStatus = {
+  WAITING: 'WAITING',
+  APPROVED: 'APPROVED',
+};
+
+let taskRequest;
+
+const taskRequestSkeleton = document.querySelector('.taskRequest__skeleton');
+const taskSkeleton = document.querySelector('.task__skeleton');
+const requestorSkeleton = document.querySelector(
+  '.requestors__container__list__skeleton',
+);
+
+const taskRequestContainer = document.getElementById('task-request-details');
+const taskContainer = document.getElementById('task-details');
+const requestorsContainer = document.getElementById('requestors-details');
+
+const taskRequestId = new URLSearchParams(window.location.search).get('id');
+history.pushState({}, '', window.location.href);
+
+let taskId;
+
+function renderTaskRequestDetails(taskRequest) {
+  taskRequestContainer.append(
+    createCustomElement({
+      tagName: 'h1',
+      textContent: `Task Request `,
+      class: 'taskRequest__title',
+      child: [
+        createCustomElement({
+          tagName: 'span',
+          class: 'taskRequest__title__subtitle',
+          textContent: `#${taskRequest.id}`,
+        }),
+      ],
+    }),
+    createCustomElement({
+      tagName: 'p',
+      textContent: 'Status: ',
+      class: 'taskRequest__status',
+      child: [
+        createCustomElement({
+          tagName: 'span',
+          textContent: taskRequest.status,
+          class: [
+            'taskRequest__status__chip',
+            `taskRequest__status__chip--${taskRequest.status.toLowerCase()}`,
+          ],
+        }),
+      ],
+    }),
+  );
+}
+
+async function renderTaskDetails(taskId) {
+  try {
+    const res = await fetch(`${API_BASE_URL}/tasks/${taskId}/details`);
+    taskSkeleton.classList.add('hidden');
+    const data = await res.json();
+
+    const { taskData } = data;
```

Comment:
The taskData should be initialised with an {} so that we can avoid undefined errors

Reply:
addressed, pls check